rust/rbac-registration/src/cardano/cip509/cip509.rs

```diff
@@ -0,0 +1,569 @@
+                    },
+                    Cip509IntIdentifier::PreviousTxId => {
+                        result.prv_tx_id =
+                            decode_previous_transaction_id(d, context, &result.report);
+                    },
+                    Cip509IntIdentifier::ValidationSignature => {
+                        result.validation_signature =
+                            decode_validation_signature(d, context, &result.report);
+                    },
+                }
+            } else {
+                // Handle the x509 chunks 10 11 12
+                // Technically it is possible to store multiple copies (or different instances) of
+                // metadata, but it isn't allowed. See this link for more details:
+                // https://github.com/input-output-hk/catalyst-CIPs/blob/x509-envelope-metadata/CIP-XXXX/README.md#keys-10-11-or-12---x509-chunked-data
+                if is_metadata_found {
+                    result.report.duplicate_field(
+                        "metadata",
+                        "Only one instance of the chunked metadata should be present",
+                        context,
+                    );
+                    continue;
+                }
+                is_metadata_found = true;
+
+                match X509Chunks::decode(d, decode_context) {
+                    Ok(chunks) => result.metadata = chunks.into(),
+                    Err(e) => {
+                        result.report.other(
+                            &format!("Unable to decode metadata from chunks: {e:?}"),
+                            context,
+                        );
+                    },
+                };
+            }
+        }
+
+        let required_keys = [
+            Cip509IntIdentifier::Purpose,
+            Cip509IntIdentifier::TxInputsHash,
+            Cip509IntIdentifier::ValidationSignature,
+        ];
+        report_missing_keys(&found_keys, &required_keys, context, &result.report);
+        if !is_metadata_found {
+            result
+                .report
+                .missing_field("metadata (10, 11 or 12 chunks)", context);
+        }
+
+        Ok(result)
+    }
+}
+
+/// Records the payment history for the given set of addresses.
+fn payment_history(
+    txn: &conway::MintedTx, track_payment_addresses: &[ShelleyAddress], origin: &PointTxnIdx,
+    report: &ProblemReport,
+) -> HashMap<ShelleyAddress, Vec<Payment>> {
+    let hash = MultiEraTx::Conway(Box::new(Cow::Borrowed(txn))).hash();
+    let context = format!("Populating payment history for Cip509, transaction hash = {hash:?}");
+
+    let mut result: HashMap<_, _> = track_payment_addresses
+        .iter()
+        .cloned()
+        .map(|a| (a, Vec::new()))
+        .collect();
+
+    for (index, output) in txn.transaction_body.outputs.iter().enumerate() {
+        let conway::PseudoTransactionOutput::PostAlonzo(output) = output else {
+            report.other("Unsupported legacy transaction output", &context);
```

Comment:
Since we are recording the payment address that we care, not sure whether we want to report the problem here, if it somehow legacy outputs, we just don't care about it. WDYT?

Reply:
I have followed the [previous logic](https://github.com/input-output-hk/catalyst-libs/blob/main/rust/rbac-registration/src/registration/cardano/mod.rs#L599), but I think that you are right. I will change that, but just in case let's ask @stevenj.